staging/src/k8s.io/component-base/logs/options.go

```diff
@@ -17,49 +17,46 @@ limitations under the License.
 package logs
 
 import (
-	"github.com/go-logr/logr"
 	"github.com/spf13/pflag"
 
 	"k8s.io/klog/v2"
 
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/component-base/config"
 	"k8s.io/component-base/logs/sanitization"
 )
 
 // Options has klog format parameters
 type Options struct {
-	LogFormat       string
-	LogSanitization bool
+	Config config.LoggingConfiguration
 }
 
 // NewOptions return new klog options
 func NewOptions() *Options {
```

Comment:
Setting up defaults like this means we break compatibility with consumers that embed the log config options.

The expectation is that any consuming component that embeds the log config struct will get stable defaults for the duration of that version.

The normal approach is to:

1. define API defaults for the config struct, which are versioned (e.g. pkg/scheduler/apis/config/v1beta1/defaults.go)
2. construct default options by creating a versioned config struct and applying defaults (e.g. cmd/kube-scheduler/app/options/options.go#newDefaultComponentConfig())

Reply:
Ok, looks like still some components (for example kube-apiserver) don't use config structs. This means that we will still need to maintain Options as a way to configure logging flags.

For components that allow using Configs, like kube-scheduler you mentioned, I will created dedicated PRs to migrate them.